exec/java-exec/src/main/java/org/apache/drill/exec/store/dfs/FileSelection.java

```diff
@@ -213,14 +214,15 @@ public static FileSelection create(final List<FileStatus> statuses, final List<S
     if (statuses == null || statuses.isEmpty()) {
       selectionRoot = commonPathForFiles(files);
     } else {
-      if (statuses.size() == 1 && !Strings.isNullOrEmpty(root)) {
-        final Path rootPath = new Path(root);
-        final URI uri = statuses.get(0).getPath().toUri();
-        final Path path = new Path(uri.getScheme(), uri.getAuthority(), rootPath.toUri().getPath());
-        selectionRoot = path.toString();
-      } else {
-        selectionRoot = commonPath(statuses);
+      assert !Strings.isNullOrEmpty(root);
```

Comment:
What is the expected behavior when root is null and assertions are disabled? This throws an NPE, I think we should not.